src/socket.c

```diff
@@ -54,20 +61,27 @@
 	} \
 } while(0)
 
-#define PTHREADS_SOCKET_ERROR() do { \
-	int eno = php_socket_errno(); \
-	char *estr = eno != SUCCESS ? \
-		php_socket_strerror(eno, NULL, 0) : \
-		NULL; \
-	zend_throw_exception_ex(spl_ce_RuntimeException, eno, \
-		"Error (%d): %s", eno, estr ? estr : "unknown"); \
-	if (eno != SUCCESS) { \
-		if (estr) { \
-			efree(estr); \
+#define PTHREADS_HANDLE_SOCKET_ERROR(eno) do { \
+	if ((eno) != EAGAIN && (eno) != EWOULDBLOCK && (eno) != EINPROGRESS && (eno) != SOCK_EINVAL) { \
```

Comment:
`EWOULDBLOCK` was written to be conditionally used lower down. Will it always be available? If not, then it should be conditionally used here too.

Reply:
On modern unix systems with GNU C library `EWOULDBLOCK` is always available with same value of EAGAIN. On modern windows systems `EWOULDBLOCK` is available but Microsoft do not want to guarantee `EWOULDBLOCK`. I've changed that. If error no `EWOULDBLOCK` doesn't exist on win32, it will be defined.